### PR TITLE
Move to new io/path/fs/net API

### PR DESCRIPTION
**Pull request by panicbit**
- [x] ~~Requires reem/iron-test#13 to get merged~~
- [x] ~~Requires iron/iron#303 to get merged~~
- [x] Needs updated version of reem/iron-test at crates.io

### CODE REVIEW

**Comment by reem** [comment]
Updated iron-test.

File: examples/doc_server.rs [hunk 1]
```diff
@@ -27,5 +28,5 @@ fn main() {
 
     println!("Doc server running on http://localhost:3000/doc/");
 
-    Iron::new(mount).listen((Ipv4Addr(127, 0, 0, 1), 3000)).unwrap();
+    Iron::new(mount).http((IpAddr::new_v4(127, 0, 0, 1), 3000)).unwrap();
```
Comment (reem) [inline examples/doc_server.rs]: You can just use `"localhost:3000"` instead of this whole `IpAddr` thing

Comment (panicbit) [inline examples/doc_server.rs]: I thought the same but I didn't want to diverge too much. I'll change it now.

**Comment by panicbit** [comment]
@reem Shall I squash now?

**Comment by reem** [comment]
Yup, go ahead and squash.

**Comment by panicbit** [comment]
@reem Closes #28

**Pull request merged**